contrib/custom_templates.go

```diff
@@ -19,16 +19,23 @@ const (
 	SetCustomTemplatePath = "sets/sets.gotmpl"
 )
 
-var Sets = func() model.CustomTemplates {
-	templateContents, err := templatesBox.FindString(SetCustomTemplatePath)
+func init() {
+	sets, err := Sets()
 	if err != nil {
 		panic(err)
 	}
+	AllCustomTemplates = append(AllCustomTemplates, sets)
+}
+
+func Sets() (model.CustomTemplates, error) {
+
+	templateContents, err := templatesBox.FindString(SetCustomTemplatePath)
+	if err != nil {
+		return model.CustomTemplates{}, err
+	}
 	setsTemplates := model.CustomTemplates{
 		Templates: map[string]string{SetOutputFilename: templateContents},
 	}
-	// register sets
-	AllCustomTemplates = append(AllCustomTemplates, setsTemplates)
 
-	return setsTemplates
-}()
+	return setsTemplates, nil
+}
```

Comment:
why this change? honestly what i like about the other model is how easily copy-pasteable it is. this will require editing functions in 2 places

Reply:
The other way was a pattern I had never seen before. It was attempting to use `init()` functionality without actually using an `init()` function. I personally think this version is more immediately understandable, and closer to common go paradigms.